app/Http/Controllers/MessagesController.php

```diff
@@ -89,22 +90,38 @@ public function show($id)
      */
     public function sendMessage(Message $message)
     {
-        $contestId = request('contest_id');
+        // Get competition with activity.
         $competitionId = request('competition_id');
         $competition = Competition::find($competitionId);
-        $contest = Contest::find($contestId);
-        $contest = $this->manager->appendCampaign($contest);
 
-        $users = $this->manager->getModelUsers($competition, true);
+        // Get competition with activity from flash if it is there.
+        // Otherwise, grab it.
+        $key = generate_model_flash_session_key($competition, ['includeActivity' => true]);
 
-        // Only send checkin messages to users who haven't reported back.
-        if ($message->type === 'checkin') {
-            $users = $users->where('reportback', null);
+        if (session()->has($key)) {
+            $competition = session($key);
+        } else {
+            $competition = $this->manager->getCompetitionOverview($competition, true);
+        }
+
+        // Send test emails to authenticated user.
+        if (request('test')) {
+            $user = Auth::user();
+            $user = $this->userRepository->find($user->id);
+            $user = $this->manager->appendReportback($user, null);
+
+            $users = [$user];
+        } else {
+            $users = $competition->contestants;
+
+            // Only send checkin messages to users who haven't reported back.
+            if ($message->type === 'checkin') {
+                $users = $users->where('reportback', null);
```

Comment:
If using the `$competition` full overview instance, these users would be `$competition->activity->inactive` right?

(or `$competition->activity['inactive']`... can't remember if it's array or object